src/main/java/org/jenkinsci/plugins/workflow/actions/WarningAction.java

```diff
@@ -0,0 +1,37 @@
+package org.jenkinsci.plugins.workflow.actions;
+
+import hudson.model.Result;
+import javax.annotation.Nonnull;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+/**
+ * This action can be attached to a {@link FlowNode} to signify that some non-fatal
+ * warning occurred during execution. Visualizations should treat FlowNodes with this
```

Comment:
BTW I find it clearest to put the Javadoc summary sentence on its own line in sources, as it will be treated specially.

Reply:
Done in 7d94c66.